addons/mail_group/security/mail_group_security.xml

```diff
@@ -0,0 +1,126 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo noupdate="1">
+    <record id="mail_group_rule_all" model="ir.rule">
+        <field name="name">Mail Group: Access only public and joined groups</field>
+        <field name="model_id" ref="model_mail_group"/>
+        <field name="domain_force">[
+            '|',
+            '|',
+                ('moderator_ids', 'in', user.id),
+                ('privacy', '=', 'public'),
+                '&amp;',
+                    ('privacy', '=', 'groups'),
+                    ('group_public_id', 'in', [g.id for g in user.groups_id]),
+            ]
+        </field>
+        <field name="groups" eval="[(4, ref('base.group_user')), (4, ref('base.group_portal')), (4, ref('base.group_public'))]"/>
+        <field name="perm_create" eval="True"/>
+        <field name="perm_read" eval="True"/>
+        <field name="perm_write" eval="True"/>
+        <field name="perm_unlink" eval="True"/>
```

Comment:
Unsure for unlink for users, meaning a "standard" user could unlink a mailing list used by thousands of people. Maybe to clean (or discuss) in a future task

Reply:
Added a note in the pad about that